engine/wwtlib/HipsProperties.cs

```diff
@@ -0,0 +1,213 @@
+                //{
+                //    string tableUrl = propsUrl.Replace("/properties", "/metadata.xml");
+
+                //    client = new WebClient();
+                //    string voTable = client.DownloadString(tableUrl);
+
+                //    System.Xml.XmlDocument xmlDocument = new System.Xml.XmlDocument();
+                //    xmlDocument.LoadXml(voTable);
+                //    props.VoTable = new VoTable(xmlDocument);
+                //}
+            }
+            catch
+            {
+                props.Properties["dummy"] = "failed";
+            }
+            
+            return props;
+        }
+
+        private void Download(string propsUrl)
+        {
+            webFile = new WebFile(propsUrl);
+            webFile.OnStateChange = LoadData;
+            webFile.Send();
+        }
+
+        private void LoadData()
+        {
+            if (webFile.State == StateType.Error)
+            {
+                Script.Literal("alert({0})", webFile.Message);
+            }
+            else if (webFile.State == StateType.Received)
+            {
+                ParseProperties(webFile.GetText());
+                downloadComplete = true;
+            }
+        }
+
+        public static void ValidateProperties(HipsProperties props)
+        {
+            if (props.Properties.Count == 1)
+            {
+                return;
+            }
+            if (!props.Properties.ContainsKey("obs_title"))
+            {
+                if (props.Properties.ContainsKey("creator_did"))
+                {
+                    props.Properties["obs_title"] = props.Properties["creator_did"];
+                }
+                else
+                {
+                    props.Properties["obs_title"] = "No Title";
+                }
+            }
+        }
+
+        public static bool IsValid(HipsProperties props)
+        {
+            if (props.Properties.Count == 1)
+            {
+                return false;
+            }
+            if (!props.Properties.ContainsKey("obs_title"))
+            {
+                return false;
+            }
+            if (!props.Properties.ContainsKey("hips_tile_format"))
+            {
+                return false;
+            }
+            if (!props.Properties.ContainsKey("hips_service_url"))
+            {
+                return false;
+            }
+            if (!props.Properties.ContainsKey("hips_order"))
+            {
+                return false;
+            }
+            if (!props.Properties.ContainsKey("dataproduct_type"))
+            {
+                return false;
+            }
+            if (props.Properties["dataproduct_type"] == "cube")
+            {
+                return false;
+            }
+            if (props.Properties.ContainsKey("client_category") && props.Properties["client_category"].ToLowerCase().IndexOf("solar") > -1)
```

Comment:
Do you know what this test is all about? It seems odd to me.

Reply:
Hmm... Sure looks unnecessary to me. The function itself is not yet used, but I'll have to take a closer look when it is time to fully activate this part.